subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultFileSystemSnapshotter.java

```diff
@@ -154,7 +139,7 @@ public FileTreeSnapshot create() {
      * before any downstream task uses them.
      */
     @Override
-    public FileTreeSnapshot snapshotDirectoryTree(final DirectoryFileTree dirTree) {
+    public VisitableDirectoryTree snapshotDirectoryTree(final DirectoryFileTree dirTree) {
```

Comment:
These methods don't snapshot the trees anymore, they just create the "snapshotting visitor".